src/jit/codegenxarch.cpp

```diff
@@ -3790,15 +3784,11 @@ void CodeGen::genCodeForCpBlk(GenTreeBlk* cpBlkNode)
     GenTree* srcAddr   = nullptr;
 
     // Size goes in arg2
-    if (blockSize != 0)
+    if (cpBlkNode->gtOper != GT_STORE_DYN_BLK)
```

Comment:
FYI practically all code in this function up to `genConsumeBlockOp` is useless.

Reply:
I disagree. This code historically had a number of issues due to mismatches between `Lowering` and `CodeGen`, so perhaps there is an overabundance of asserts, but I would prefer to retain them.

Reply:
I agree with Carol, these asserts are a direct byproduct of the many issues from this code.

Reply:
Asserting that the size of block is within a certain range does not achieve anything because the generate code is not specific to certain sizes. The problem is that this whole thing is as sloppy as it gets. Case in point - this code generates a `LEA` that is a duplicate of another `LEA` that's generated by `genConsumeBlockOp`. All these asserts are actually part of the problem and not the solution.

Reply:
That may indeed be the case, and I agree that this could be improved. That said, I think that cleaning up the asserts would involve a more robust handshake between `Lowering` and `CodeGen` that's best left for a future PR.